go/metadata/capabilities/capabilities.go

```diff
@@ -143,12 +147,89 @@ func normalize(key string, value interface{}) interface{} {
 			out[k] = strings.ToLower(v.(string))
 			continue
 		}
+		if k == "noProxy" {
+			if vs, ok := v.(string); ok {
+				out["noProxy"] = strings.Split(vs, ",")
+				continue
+			}
+		}
 		out[k] = v
 	}
 
+	return key, out
+}
+
+// denormalizeW3C remove non-W3C capabilities.
+func denormalizeW3C(in map[string]interface{}) map[string]interface{} {
+	out := map[string]interface{}{}
+
+	for k, v := range in {
+		if k == "goog:chromeOptions" {
+			denormalized["chromeOptions"] = v
+		}
+		// extension capabilities
+		if strings.Contains(k, ":") {
+			out[k] = v
+			continue
+		}
+		for _, a := range w3cSupportedCapabilities {
+			if k == a {
+				out[k] = v
+				break
+			}
+		}
+	}
+
 	return out
 }
 
+func denormalizeJWP(caps map[string]interface{}) map[string]interface{} {
+	denormalized := map[string]interface{}{}
+
+	for k, v := range caps {
+		if k == "goog:chromeOptions" {
+			denormalized["chromeOptions"] = v
+		}
+
+		if k != "proxy" {
+			denormalized[k] = v
+			continue
+		}
+
+		proxy, ok := v.(map[string]interface{})
+		if !ok {
+			denormalized[k] = v
+			continue
+		}
+
+		denormalizedProxy := map[string]interface{}{}
+		for pk, pv := range proxy {
+			if pk != "noProxy" {
+				denormalizedProxy[pk] = pv
+				continue
+			}
+
+			noProxy, ok := pk.([]interface{})
```

Comment:
You should do the type assertion on pv, not pk.

Reply:
Done.